src/error/utils.rs

```diff
@@ -0,0 +1,92 @@
+use cssparser::Token;
+
+pub(crate) fn render_token<'a>(token: &Token<'a>) -> String {
+    // THIS TOOK FOREVER TO IMPLEMENT
+    // TODO: Make this easier to read, I guess
+
+    match token {
+        // TODO: Give these guys some better names
+        Token::Ident(ident) => format!("{}", ident.clone()),
+        Token::AtKeyword(value) => format!("@{}", value.clone()),
+        Token::Hash(name) | Token::IDHash(name) => format!("#{}", name.clone()),
+        Token::QuotedString(value) => format!("\"{}\"", value.clone()),
+        Token::Number {
+            has_sign: signed,
+            value: num,
+            int_value: _,
+        }
+        | Token::Percentage {
+            has_sign: signed,
+            unit_value: num,
+            int_value: _,
+        } => render_number(*signed, *num, &token),
+        Token::Dimension {
+            has_sign: signed,
+            value: num,
+            int_value: _,
+            unit,
+        } => format!("{}{}", render_int(*signed, *num), unit),
+        Token::WhiteSpace(_) => String::from(" "),
+        Token::Comment(comment) => format!("/* {} */", comment),
+        Token::Function(name) => format!("{}()", name),
```

Comment:
Maybe we should `clone()` here (lines from 30 to 33) (as you have done on line 9)